### PR TITLE
Allow turbo-modules with names with @my-org/project-name

### LINKED ISSUE #124: Name mangling breaks for iOS in organization-scoped packages
In an organization scoped-package such as `@unomed/react-native-matrix-sdk`, UBRN puts the podspec into `./@unomed/react-native-matrix-sdk.podspec`. However, the file actually needs to go into the root folder as `unomed-react-native-matrix-sdk.podspec`.

Similarly, inside the file, UBRN uses `s.name = "@unomed/react-native-matrix-sdk"` while it should be `s.name  = "unomed-react-native-matrix-sdk"`.

When I move the file, change the name and rebase the paths, everything works fine again.

PS: I also tested Android in an organization-scoped package. Everything appears to work fine out of the box there. 👌

### PR DESCRIPTION
**Pull request by jhugman**
Fixes #124 

According to [The Big O of Code Reviews](https://www.egorand.dev/the-big-o-of-code-reviews/), this is a O(_n_) change.

This adds to the tests introduced in #123 to support names with an org name.

### CODE REVIEW

File: scripts/run-checkout-tests.sh [hunk 1]
```diff
@@ -40,6 +40,7 @@ assert_eq $(git ls-remote --heads origin | grep refs/heads/releases/v1 | cut -f1
 popd
 
 clean_up
+exit 0
```
Comment (Johennes) [inline scripts/run-checkout-tests.sh:43]: I think this might be a leftover?

File: scripts/test-turbo-modules.sh [hunk 2]
```diff
@@ -401,6 +402,51 @@ run_default() {
...
   # ReactNativeDummyLib fails with "› Must be a valid npm package name"
+  main \
+    --force-new-directory \
+    --keep-directory-on-exit \
+    --ubrn-config "$config" \
+    --builder-bob-version 0.35.1 \
+    --skip-ios \
+    --skip-android \
+    --slug @my-org/react-native-dummy-lib \
+    "$working_dir/@my-org/react-native-dummy-lib"
```
Comment (Johennes) [inline scripts/test-turbo-modules.sh:413]: The lines below use different indentation. Not that it matters at all of course. 😅

```suggestion
      --force-new-directory \
      --keep-directory-on-exit \
      --ubrn-config "$config" \
      --builder-bob-version 0.35.1 \
      --skip-ios \
      --skip-android \
      --slug @my-org/react-native-dummy-lib \
      "$working_dir/@my-org/react-native-dummy-lib"
```

File: crates/ubrn_cli/src/codegen/mod.rs [hunk 3]
```diff
@@ -327,7 +327,7 @@ mod files {
     templated_file!(PodspecTemplate, "module-template.podspec");
     impl RenderedFile for PodspecTemplate {
         fn path(&self, project_root: &Utf8Path) -> Utf8PathBuf {
-            let name = self.config.project.raw_name();
+            let name = self.config.project.cpp_filename();
```
Comment (Johennes) [inline crates/ubrn_cli/src/codegen/mod.rs:330]: It strikes me as slightly odd that the filename for C++ is reused for the podspec. It doesn't really make a huge difference if the value is the same but maybe using a dedicated function would be cleaner?

**Review by Johennes — COMMENTED** [review]
I tested this on https://github.com/unomed-dev/react-native-matrix-sdk using the example app and things work fine for iOS. On Android, the app appears to crash. At least I think it does. When I tap the button to make the native call in the app, it looks as if it goes into the background unexpectedly but then seems to relaunch entirely when I try to foreground it. I haven't seen this behaviour before and I'm not getting _any_ related output on the terminal. 🤔

**Comment by zzorba** [comment]
I'll test out this build on my side @Johennes to see if I get similar results.  I don't have a namespaced package like this, but I'm guessing the crash you are seeing is unrelated to that?

**Comment by Johennes** [comment]
> I'll test out this build on my side @Johennes to see if I get similar results. I don't have a namespaced package like this, but I'm guessing the crash you are seeing is unrelated to that?

Possibly though I'm not entirely sure since the organisation ends up in some generated names but not others. I'll try to also give it some more testing tomorrow.

**Comment by Johennes** [comment]
Yeah, I noticed that, too, but already had `index` without any extension in `package.json` so didn't have to change anything. I think I'll try running the app from Android Studio later today to see that gives me an actual error message or stack trace.

File: scripts/run-checkout-tests.sh (hunk 1 above)
Comment (jhugman) [reply scripts/run-checkout-tests.sh]: Yes! That was trying to get the error messages for #126 !

Removed now.

**Comment by jhugman** [comment]
> > I'll test out this build on my side @Johennes to see if I get similar results. I don't have a namespaced package like this, but I'm guessing the crash you are seeing is unrelated to that?
> 
> Possibly though I'm not entirely sure since the organisation ends up in some generated names but not others. I'll try to also give it some more testing tomorrow.

I found it!

The Android crash was due to the library name in `System.loadLibrary` not matching the one built in `CMakeLists.txt`. 

I think I broke it in #123, but might have broken it in this one.

I've changed this and added a test.

**Comment by jhugman** [comment]
> One thing I noticed here, it seems like it went from `src/index.ts` to `src/index.tsx`, so I had to update my package.json slightly.

Our new tests check for differences between the generated files. Builder bob was generating an `index.tsx` file, and `ubrn` was generating an `index.ts` file.

**Comment by zzorba** [comment]
Confirmed the fix is working!

**Review by zzorba — APPROVED** [review]
LGTM

**Pull request merged**